src/main/webapp/app/exercises/shared/result/result.service.ts

```diff
@@ -31,12 +37,78 @@ export class ResultService implements IResultService {
     private resultResourceUrl = SERVER_API_URL + 'api/results';
     private participationResourceUrl = SERVER_API_URL + 'api/participations';
 
-    constructor(private http: HttpClient) {}
+    private readonly maxValueProgrammingResultInts = 255; // Size of tinyInt in SQL, that is used to store these values
+
+    constructor(private http: HttpClient, private translateService: TranslateService) {}
 
     find(resultId: number): Observable<EntityResponseType> {
         return this.http.get<Result>(`${this.resultResourceUrl}/${resultId}`, { observe: 'response' }).pipe(map((res: EntityResponseType) => this.convertDateFromServer(res)));
     }
 
+    getResultString(result: Result | undefined, exercise: Exercise | undefined): string {
+        if (result && exercise) {
+            return this.getResultStringNotUndefined(result, exercise);
+        } else {
+            captureException('Tried to generate a result string, but either the result or exercise was undefined');
```

Comment:
Benedikt suggested the exact opposite: https://github.com/ls1intum/Artemis/pull/5175#discussion_r902555716
Since we have many components where the objects like result or exercise are marked as optional, I prefer being on the safe side and simply show an empty string instead of actually getting an error in the client.
The components should make sure to actually pass the values, but if there is a bug somewhere, this behaviour is better imho than getting an error